### PR TITLE
Fix #6418 #7248 #7360 Label attribute fixes

### LINKED ISSUE #6418: As data manager user I require that 'label' attribute cannot be nillable
### How to Reproduce
Create a table with a label attribute that is nillable

### Expected behavior
When I select this attribute in metadata manager I get an error message.
(I expect it to be selectable otherwise users don't get it).

### Observed behavior
Is accepted. As a consequence these attributes are invisible in mapping service.

### PR DESCRIPTION
**Pull request by tommydeboer**
Fixes:
#6418 As data manager user I require that 'label' attribute cannot be nillable
#7248 Label attribute can be hidden 
#7360 Entities without labels cannot be saved in the metadata manager

#### Checklist
- [x] Functionality works & meets specifications
- [x] Code reviewed
- [x] Code unit/integration/system tested
- [x] User documentation updated
- [x] (If you have changed REST API interface) view-swagger.ftl updated
- [x] Test plan template updated
- [x] Clean commits
- [ ] Added Feature/Fix to release notes

### CODE REVIEW

**Comment by fdlk** [comment]
test this please

**Comment by fdlk** [comment]
test this please

**Comment by fdlk** [comment]
test this please

**Comment by fdlk** [comment]
test this please

**Comment by fdlk** [comment]
test this please

**Comment by fdlk** [comment]
test this please

**Comment by fdlk** [comment]
test this please

**Comment by tommydeboer** [comment]
test this please

File: molgenis-api-tests/src/test/java/org/molgenis/api/tests/utils/RestTestUtils.java [hunk 1]
```diff
@@ -18,6 +18,7 @@
 
 import static com.google.common.collect.ImmutableMap.of;
 import static io.restassured.RestAssured.given;
+import static java.lang.String.format;
```
Comment (fdlk) [inline molgenis-api-tests/src/test/java/org/molgenis/api/tests/utils/RestTestUtils.java]: unneeded x2

File: molgenis-data-validation/src/main/java/org/molgenis/data/validation/meta/EntityTypeValidator.java [hunk 2]
```diff
@@ -146,7 +160,15 @@ static void validateOwnLabelAttribute(EntityType entityType, Map<String, Attribu
...
-						format("Label attribute [%s] is not is not one of the attributes of entity [%s]",
+						format("Label attribute [%s] is not one of the attributes of entity [%s]",
+								ownLabelAttr.getName(), entityType.getId())));
+			}
+
+			// Validate that the label attribute is not nillable
+			if (ownAttr.isNillable())
+			{
+				throw new MolgenisValidationException(new ConstraintViolation(
+						format("Label attribute [%s] of entity type [%s] is not a non-nillable attribute",
```
Comment (fdlk) [inline molgenis-data-validation/src/main/java/org/molgenis/data/validation/meta/EntityTypeValidator.java]: not non nillable -> nillable

File: molgenis-data/src/main/java/org/molgenis/data/meta/model/Attribute.java [hunk 3]
```diff
@@ -199,6 +199,10 @@ public boolean isLabelAttribute()
 	public Attribute setLabelAttribute(Boolean isLabelAttr)
 	{
 		set(IS_LABEL_ATTRIBUTE, isLabelAttr);
+		if (isLabelAttr != null && isLabelAttr)
```
Comment (fdlk) [inline molgenis-data/src/main/java/org/molgenis/data/meta/model/Attribute.java]: We ususally don't put the logic in the entity class. Will be confusing that this setter has side effects:
`addAttribute("blah").setNillable(true).setLabelAttribute(true)`

File: molgenis-metadata-manager/src/main/resources/l10n/metadata-manager_en.properties [hunk 4]
```diff
@@ -74,20 +66,19 @@ attribute-edit-form-maximum-range-placeholder=Add a maximum range...
...
 confirm-before-leaving-text=All unsaved changes will be lost, are you sure you want to continue?
 confirm-before-leaving-confirm=Yes
 confirm-before-leaving-cancel=No, stay
-
 # actions
-save-succes-message=Successfully updated metadata for EntityType
\ No newline at end of file
+save-succes-message=Successfully updated metadata for EntityType
+# toast
+toast-id-required=ID attribute can not be empty
```
Comment (fdlk) [inline molgenis-metadata-manager/src/main/resources/l10n/metadata-manager_en.properties]: can not -> cannot 2x

**Comment by fdlk** [comment]
#7360 is not completely fixed yet for me, it refuses to save an *abstract* entity with invisible ID attribute and without label.

**Comment by tommydeboer** [comment]
> #7360 is not completely fixed yet for me, it refuses to save an abstract entity with invisible ID attribute and without label.
I removed the client-side checks and then I run into server-side checks which complain about the same.

I'm tempted to say this is a different issue: the validation rules of the Metadata Manager are incorrect. You can't make an abstract entity type without an identifier either, for example. I'll have a quick look though, maybe it's easy to fix.

**Pull request merged**